src/Helpers/Form/Template.php

```diff
@@ -111,7 +111,7 @@ public static function handleOptionsBackwardCompatibility($settings)
                 $settings['payment_amount']['decimals_enabled'] = $settings['visual_appearance']['decimals_enabled'];
             }
             $settings['introduction']['primary_color'] = $settings['visual_appearance']['primary_color'];
-        } elseif (isset($settings['payment_amount'], $settings['introduction'])) {
+        } elseif (isset($settings['payment_amount']['decimals_enabled'], $settings['introduction']['primary_color'])) {
```

Comment:
@LMNTL I reviewed your pull request. It will be fixed notice but  It can cause unexpected output because of the missing value of these settings.
I want to suggest you keep `elseif` condition as is. Update logic inside `elseif`, if the setting value does not exist then set it to the default value of the field.

Let me know if you have any question.

Reply:
@ravinderk Good catch! I've refactored the logic inside the `elseif`. I also made sure to set the default value based on the current form template. When you get a chance, could you let me know if it needs any more revisions?